README.md

```diff
@@ -5,757 +5,875 @@
+
+let number = create(Number, 42).set(43);
+
+number.valueOf()
+//> 43
+```
+
+You can also use `set` transition to replace the current Microstate with another Microstate. This is especially useful when building state machines because it allows you to change the type of the current Microstate. By changing the type, you're also changing available transitions and how the state is calculated.
 
 ```js
-class AuthenticatedSession extends Session {
-  isAuthenticated = true;
-  content = Object;
+import { types } from 'microstates';
 
-  logout() {
-    return create(AnonymousSession);
+class Vehicle {
+  weight = Number;
+  towing = Vehicle;
+
+  get isTowing {
+    return !!this.towing;
+  }
+
+  get towCapacity() {
+    throw new Error('not implemented');
+  }
+
+  tow() {
+    throw new Erorr('not implemented');
   }
 }
 
-class AnonymousSession extends Session {
-  content = null;
-  isAuthenticated = false;
-  authenticate(user) {
-    return create(AuthenticatedSession, { content: user });
+class Car extends Vehicle {}
+
+class Truck extends Vehicle {
+  towCapacity = Number;
+
+  tow(vehicle) {
+    if (vehicle.weight.state < this.towCapacity.state) {
+      return this.towing.set(vehicle);
+    } else {
+      throw new Error(`Unable to tow ${vehicle.weight.state} because it exceeds tow capacity of ${this.towCapacity.state}`);
+    }
   }
 }
 
-class MyApp {
-  session = Session;
+class Friend {
+  vehicle = Vehicle;
 }
 
-// without initial state it initializes into AnonymousSession state
-create(MyApp).state;
-// => { session: { content: null, isAuthenticated: false }}
+let prius = create(Car, { weight: 3000 });
+
+let mustang = create(Car, { weight: 3500, towCapacity: 1000 });
+let f150 = create(Truck, { weight: 5000, towCapacity: 13000 });
+
+let rob = create(Friend).vehicle.set(mustang);
+let charles = create(Friend).vehicle.set(f150);
 
-// transition to AuthenticatedSession state with authenticate
-create(MyApp).authenticate({ name: 'Taras' }).state;
-// => { session: { content: { name: 'Taras' }, isAuthenticated: true }};
+rob.vehicle.tow(prius);
+//> Error: not implemented
 
-// restore into AuthenticatedSession state
-create(MyApp, { session: { content: { name: 'Taras' } } }).state;
-// => { session: { content: { name: 'Taras' } }, isAuthenticated: true }
+let result = charles.vehicle.tow(prius)
 
-// transition to AnonymousSession state with logout
-create(MyApp, { session: { content: { name: 'Taras' } } }).logout().state;
-// => { session: { content: null, isAuthenticated: false }}
+result.vehicle.state.isTowing
+//> true
 ```
 
-## Parameterized Arrays & Objects
+Those familiar with functional programming might recognize this as a flatMap operation. It is not required for you to understand Monads to use Microstates transitions. If you're interested in learning about the primitives of functional programming, you may checkout [funcadelic.js](https://github.com/cowboyd/funcadelic.js). Microstates uses Funcadelic under the hood.
```

Comment:
Nice shout out! Should we make this another "pro tip:"?

This is also an opportunity to express our core value of pragmatism and not letting the FP details "leak" into our implomentation.

> Pro tip: Microstates will never require you to understand Monads in order to use transitions, but if you're interested in learning about the primitives of functional programming that power Microstates, you may want to checkout [funcadelic.js](https://github.com/cowboyd/funcadelic.js).